core/msg.c

```diff
@@ -370,6 +370,25 @@ static int _msg_receive(msg_t *m, int block)
     DEBUG("This should have never been reached!\n");
 }
 
+int msg_avail(void)
+{
+    unsigned state = disableIRQ();
+    DEBUG("msg_available: %" PRIkernel_pid ": msg_available.\n",
+          sched_active_thread->pid);
+
+    tcb_t *me = (tcb_t*) sched_threads[sched_active_pid];
```

Comment:
I was primarily talking about the existing functions which use `sched_threads[sched_active_pid]`, since it depends on where the function is used if there will be any possible races or not. I guess @Kijewski might have an opinion here?